### PR TITLE
Add twitter engine

### PR DESCRIPTION
**Pull request by liimee**
## What does this PR do?

Adds the twitter (user + post search) engine

## Why is this change important?

this would be a nice addition imo, especially since twitter is a huge platform

## How to test this PR locally?

search `!tw elon musk`

## Author's checklist

I don't think this will return good results, but it works :man_shrugging:

### CODE REVIEW

**Comment by dalf** [comment]
LGTM, thank you!

It is possible to add thumbnails but the layout is not great (the thumbnails should be on the right side, side for the news, etc..., it can be improved in a second time):
```python
        img_src = tweet.get('extended_entities', {}).get('media', [{}])[0].get('media_url_https')
        if img_src:
            img_src += "?name=thumb"
        results.append(
            {
                'url': 'https://twitter.com/i/web/status/' + tweet['id_str'],
                'title': (text[:40] + '...') if len(text) > 40 else text,
                'content': text[display[0] : display[1]],
                'img_src': img_src,
                'publishedDate': datetime.strptime(tweet['created_at'], '%a %b %d %H:%M:%S %z %Y'),
            }
        )
```

**Review by dalf — APPROVED** [review]

**Comment by liimee** [comment]
why not use `'thumbnail': img_src` instead?

File: searx/engines/twitter.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+def request(query, params):
+    params['url'] = search_url.format(url=url, query=urlencode({'q': query}))
+
+    params['headers'] = {
+        # https://github.com/zedeus/nitter/blob/5f31e86e0e8578377fa7d5aeb9631bbb2d35ef1e/src/consts.nim#L5
+        'Authorization': (
+            "Bearer AAAAAAAAAAAAAAAAAAAAAPYXBAAAAAAACLXUNDekMxqa8h%2F40K4moUkGsoc%3DTYfbDKb"
+            "T3jJPCEVnMYqilB28NHfOPqkca3qaAxGfsyKCs0wRbw"
+        )
+    }
```
Comment (return42) [inline searx/engines/twitter.py]: @liimee beside the link to nitter sources, can you explain Twitter’s Bearer authentication token a little?

Comment (liimee) [reply searx/engines/twitter.py]: I believe that's what twitter.com uses, but I don't know anything beyond that. will add it

Comment (dalf) [reply searx/engines/twitter.py]: In a private browser window (so without being logged), with the network console opened, go to https://twitter.com/searxng or any twitter URL).

Display only the XHR requests, for example the one to `https://twitter.com/i/api/1.1/attribution/event.json`.
In the sent HTTP headers, you will see the `authorization` header.

The value of token has never changed as far I know (or may be once a long time ago).

Without this header, the API doesn't work.

I realize this is not the value from Browser which starts with `AAAAAAAAAAAAAAAAAAAAANRIL` and this value was used by nitter 7 months ago ( https://github.com/zedeus/nitter/blame/79b98a80817f91e1c4cd76bcb97a15b52b885f0e/src/consts.nim ).
I'm not sure why, all I know is that this value rarely changes

**Review by dalf — APPROVED** [review]
LGTM

**Pull request merged**